### ISSUE TITLE
OffsetSampleProvider output into byte array

### ISSUE THREAD

**Issue by mimeie** [naudio/NAudio#1099]
Hi

I'm using Skip and Take successfully and I really like this feature:
```
var source = new AudioFileReader("example.mp3");
var trimmed = new OffsetSampleProvider(source) {
    SkipOver = TimeSpan.FromSeconds(30),
    Take = TimeSpan.FromSeconds(60),
WaveFileWriter.CreateWaveFile16(outputFilePath, trimmed););
```
as explained [here](https://github.com/naudio/NAudio/blob/master/Docs/OffsetSampleProvider.md)

For my case it would really help a lot if the source and the outpupt could be a byte array. So without saving it, loading it as byte array and then deleting it from the drive.

is this possible?

Thanks

**Comment by markheath** [comment]
Yes - you can keep both ends in memory, no disk involved. Wrap the input bytes in a stream-backed reader, and write the output WAV into a `MemoryStream` instead of a file.

If your input byte array is raw PCM, use `RawSourceWaveStream` (give it the matching `WaveFormat`); for the output, convert to 16-bit and write into a `MemoryStream`:

```csharp
// INPUT: raw PCM bytes -> a reader, no file
var waveFormat = new WaveFormat(44100, 16, 2);          // match your PCM data
var source = new RawSourceWaveStream(new MemoryStream(inputBytes), waveFormat);

var trimmed = new OffsetSampleProvider(source.ToSampleProvider())
{
    SkipOver = TimeSpan.FromSeconds(30),
    Take = TimeSpan.FromSeconds(60),
};

// OUTPUT: write a WAV into a MemoryStream and grab the bytes
using var ms = new MemoryStream();
WaveFileWriter.WriteWavFileToStream(ms, trimmed.ToWaveProvider16());
byte[] outputBytes = ms.ToArray();
```

`ToWaveProvider16()` does the 16-bit conversion, and `WriteWavFileToStream` writes a complete WAV (header included) into the `MemoryStream` and leaves it open, so `ms.ToArray()` gives you the file as a byte array.

Two notes:
- Your example used an `.mp3`, and `RawSourceWaveStream` is only for *raw PCM* bytes. If your input bytes are compressed MP3, swap the first line for `new Mp3FileReader(new MemoryStream(inputBytes))` (or `StreamMediaFoundationReader`) — everything else stays the same.
- If you want just raw PCM bytes rather than a WAV (no header), read from `trimmed.ToWaveProvider16()` in a loop into the `MemoryStream` instead of using `WriteWavFileToStream`.
